services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/submitSidechainCrossChainUpdate.js

```diff
@@ -48,22 +55,48 @@ const applyTransaction = async (blockHeader, tx, events, dbTrx) => {
 	};
 
 	await blockchainAppsTable.upsert(appInfo, dbTrx);
+	await ccuTable.upsert({
+		height: tx.height,
+		sendingChainID: tx.params.sendingChainID,
+		transactionID: tx.id,
+	}, dbTrx);
+
 	logger.debug(`Indexed cross chain update transaction ${tx.id} contained in block at height ${tx.height}.`);
 };
 
 const revertTransaction = async (blockHeader, tx, events, dbTrx) => {
 	if (tx.executionStatus !== TRANSACTION_STATUS.SUCCESS) return;
 
 	const blockchainAppsTable = await getBlockchainAppsTable();
+	const transactionsTable = await getTransactionsTable();
+	const ccuTable = await getCCUTable();
 
 	logger.trace(`Reverting cross chain update transaction ${tx.id} contained in block at height ${tx.height}.`);
 
+	const searchParams = {
+		sendingChainID: tx.params.sendingChainID,
+		propBetweens: [{
+			property: 'height',
+			lowerThan: blockHeader.height,
+		}],
+		sort: 'height:desc',
+		limit: 1,
+	};
+
+	const [{ height: prevTransactionHeight }] = await ccuTable.find(searchParams, 'height');
+	const [prevTransaction] = await transactionsTable.find(
+		{
+			height: prevTransactionHeight,
+			moduleCommand: tx.moduleCommand,
+			limit: 1,
```

Comment:
On the mainchain, a single block can contain multiple `.submitSideChainCrossChainUpdate` transactions from various `sendingChainIDs`. Also, we need to ensure that the relevant `prevTx.executionStatus` was success else fetch even older relevant transaction.

Ideally with `if (tx.executionStatus !== TRANSACTION_STATUS.SUCCESS) return;` within the hooks, the above shouldn't be an issue but it's better to cover any possible failure scenarios to make the code robust.